File: samplecode/match.rb
#!/usr/bin/env ruby -W0
# -*- coding: utf-8 -*-

require "rubygems"
require "opencv"
include OpenCV

result_window = GUI::Window.new "result"
temp_window = GUI::Window.new "temp"

img1 = IplImage.load(ARGV[0], CV_LOAD_IMAGE_GRAYSCALE)
img2 = IplImage.load(ARGV[1], CV_LOAD_IMAGE_GRAYSCALE)
out = img1.clone()

## mat.match_template(templ, method)
begin
  result = img1.match_template(img2, CV_TM_CCOEFF_NORMED)
  result_window.show result
  GUI::wait_key
  min_val, max_val, min_loc, max_loc = result.min_max_loc
  img1.rectangle!(CvPoint.new(max_loc.x, max_loc.y), CvPoint.new(max_loc.x + img2.cols, max_loc.y + img2.rows),
                 :color => CvColor::White, :thickness => -1)
  out.rectangle!(CvPoint.new(max_loc.x, max_loc.y), CvPoint.new(max_loc.x + img2.cols, max_loc.y + img2.rows),
                 :color => CvColor::Black, :thickness => 3)
  puts "val = #{max_val}"
  temp_window.show out
  GUI::wait_key
end while (max_val > 0.6)


temp_window.show out
GUI::wait_key

File: samplecode/match.cpp
#include <stdio.h>
#include <opencv2/core/core.hpp>
#include <opencv2/imgproc/imgproc.hpp>
#include <opencv2/highgui/highgui.hpp>

using namespace cv;

int main(int argc, char** argv)
{
  Mat img1 = imread(argv[1], CV_LOAD_IMAGE_GRAYSCALE);
  Mat img2 = imread(argv[2], CV_LOAD_IMAGE_GRAYSCALE);
  Mat out = img1.clone();
  Mat result;
  Point pt;
  double val;

  do {
    matchTemplate(img1, img2, result, CV_TM_CCOEFF_NORMED);
    imshow("result", result);
    waitKey(0);
    minMaxLoc(result, NULL, &val, NULL, &pt);
    Rect rect(pt.x,pt.y,img2.cols,img2.rows);
    rectangle(img1, rect, Scalar(0,0,255), CV_FILLED);
    rectangle(out, rect, Scalar(0,0,255), 3);
    printf("val = %f\n", val);
    imshow("temp", out);
    waitKey(0);
  } while (val > 0.6);


  //matchTemplate(img1, img2, result, CV_TM_CCOEFF_NORMED);
  //minMaxLoc(result, NULL, &val, NULL, &pt);
  //Rect rect2(pt.x,pt.y,img2.cols,img2.rows);
  //rectangle(img1, rect2, Scalar(0,0,255), 3);

  //printf("val=%f\n", val);
  imshow("temp", out);
  waitKey(0);

  return 0;
}
